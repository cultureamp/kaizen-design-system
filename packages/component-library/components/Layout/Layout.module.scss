@import "./styles";
@import "~@kaizen/component-library/styles/utils";

.root {
  @extend %root;
}

.navigationBar {
  @extend %navigation-bar;
  box-sizing: border-box;
  top: 0;
  z-index: $ca-z-index-fixed;
}

.page {
  @extend %page;
  box-sizing: border-box;
  position: relative;
}

.header {
  @extend %header;
  box-sizing: border-box;
}

.body {
  @extend %body;
  box-sizing: border-box;
}

.bodyInner {
  @extend %body-inner;
}

.sidebar {
  @extend %sidebar;
  box-sizing: border-box;
}

.content {
  @extend %content;
  box-sizing: border-box;
}

.footer {
  @extend %footer;
  box-sizing: border-box;
}

.toasts {
  position: absolute;
  top: 0;
  right: 0;
}

.announcers {
  @include sr-only();
}
