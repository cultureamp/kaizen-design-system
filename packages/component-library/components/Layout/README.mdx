---
title: Layout (component-library)
navTitle: Layout (component-library)
deprecated: true
deprecationDate: TBA
deprecationMessage: Use `@kaizen/draft-page-layout` instead.
excludeFromSidebar: true
---