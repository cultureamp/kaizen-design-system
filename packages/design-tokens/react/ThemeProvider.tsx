import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
