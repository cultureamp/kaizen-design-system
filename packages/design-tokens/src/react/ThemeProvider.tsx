import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
