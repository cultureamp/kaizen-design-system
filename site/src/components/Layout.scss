@import "~@kaizen/component-library/components/NavigationBar/styles";
@import "~@kaizen/component-library/styles/type";
@import "~@kaizen/component-library/styles/color";
@import "~@kaizen/component-library/styles/responsive";
@import "~@kaizen/component-library/styles/layers";

:root {
  --content-max-width: calc(var(--ca-grid) * 57);
}

.root {
  display: grid;
  min-height: 100vh;
  grid-template-areas:
    "navbar"
    "page"
    "footer";
  grid-template-columns: 1fr;
  grid-template-rows: [navigationBarContainer] $ca-navigation-bar__height [page] 1fr [footer];
}

.navigationBarContainer {
  grid-area: navbar;
  height: $ca-navigation-bar__height;
  z-index: $ca-z-index-fixed;
}

.page {
  display: grid;
  grid-area: page;
  grid-template-rows: [pageHeaderContainer] auto [contentContainer] 1fr;
  padding-bottom: $ca-grid * 3;
}

.noPageHeader {
  grid-template-rows: [contentContainer] 1fr;
}

.noBottomPadding {
  padding-bottom: 0;
}

.contentContainer {
  display: grid;
  justify-items: center;
  grid-template-columns: $ca-grid [content] 1fr $ca-grid;

  &.trimBottomOfCardToContent {
    grid-template-rows: min-content;
  }
}

.content {
  display: grid;
  grid-column: content;
  max-width: var(--content-max-width);
  width: 100%;
  padding: 0 0 $ca-grid;
  box-sizing: border-box;
}

.footerContainer {
  grid-area: footer;
}
