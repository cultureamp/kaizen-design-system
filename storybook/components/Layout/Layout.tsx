import React, { ReactNode } from "react"
import styles from "./Layout.module.scss"

export const Main = ({ children }: { children: ReactNode }): JSX.Element => (
  <main className={styles.main}>{children}</main>
)

export const Sidebar = ({ children }: { children: ReactNode }): JSX.Element => (
  <aside className={styles.sidebar}>{children}</aside>
)

export const Content = ({ children }: { children: ReactNode }): JSX.Element => (
  <div className={`${styles.content} tocbot-content`}>{children}</div>
)
