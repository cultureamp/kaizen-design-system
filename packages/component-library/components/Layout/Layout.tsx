import * as React from "react"

const styles = require("./Layout.module.scss")

const NavigationBar: React.FunctionComponent = ({ children }) => (
  <div className={styles.navigationBar}>{children}</div>
)

NavigationBar.displayName = "NavigationBar"

const Sidebar: React.FunctionComponent = ({ children }) => (
  <div className={styles.sidebar}>
    <div className={styles.sidebarInner}>{children}</div>
  </div>
)

Sidebar.displayName = "Sidebar"

const Header: React.FunctionComponent = ({ children }) => (
  <aside className={styles.header}>{children}</aside>
)

Header.displayName = "Header"

const Footer: React.FunctionComponent = ({ children }) => (
  <footer className={styles.footer}>{children}</footer>
)

Footer.displayName = "Footer"

/**
 * An area for toast notifications that will also trigger a screen-reader announcement.
 * Content is absolutely positioned in the top right, but it is up to you to add appropriately styled notifications.
 * By setting the children of `<Layout.Toasts>` the screen reader will immediately read its contents, without losing
 * focus. You can safely add and remove toasts without worrying about the screen reader announcement being interrupted
 * or repeated.
 */
const Toasts: React.FunctionComponent = ({ children }) => (
  <div className={styles.toasts} aria-live="assertive">
    {children}
  </div>
)

Toasts.displayName = "Toasts"

/**
 * Announcements intended for screen readers only. Content will be invisible for sighted users.
 * By setting the children of `<Layout.Announcers>` the screen reader will immediately read its contents,
 * without losing focus. You can safely override the contents when adding a new announcement rather than
 * appending the contents.
 */
const Announcers: React.FunctionComponent = ({ children }) => (
  <div className={styles.announcers} aria-live="assertive">
    {children}
  </div>
)

Announcers.displayName = "Announcers"

class Layout extends React.Component {
  static displayName = "Layout"
  static NavigationBar = NavigationBar
  static Sidebar = Sidebar
  static Header = Header
  static Footer = Footer
  static Toasts = Toasts
  static Announcers = Announcers
  render() {
    const content = React.Children.toArray(this.props.children)
    const navbar = extractChildOfType(content, NavigationBar)
    const header = extractChildOfType(content, Header)
    const sidebar = extractChildOfType(content, Sidebar)
    const footer = extractChildOfType(content, Footer)
    const announcers = extractChildOfType(content, Announcers)
    const toasts = extractChildOfType(content, Toasts)

    return (
      <div className={styles.root}>
        {navbar}
        {announcers}
        <div className={styles.page}>
          {header}
          {toasts}
          <div className={styles.body}>
            <div className={styles.bodyInner}>
              {sidebar}
              <main className={styles.content}>{content}</main>
            </div>
          </div>
          {footer}
        </div>
      </div>
    )
  }
}

const extractChildOfType = (
  children: React.ReactNode[],
  type: React.FunctionComponent<{}>
) => {
  const match = children.find(child => {
    if (React.isValidElement(child) && typeof child.type === "function") {
      return (child.type as React.SFC).displayName === type.displayName
    }

    return false
  })

  if (match) {
    const index = children.indexOf(match)
    children.splice(index, 1)
  }

  return match
}

export default Layout
