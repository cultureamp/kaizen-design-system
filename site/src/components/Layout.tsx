/* !!! This component is deprecated. Please do not use for new code  !!! */

import classnames from "classnames"
import * as React from "react"
import "../styles/global.scss"
import Footer from "./Footer"
import Head from "./Head"
import MainNav from "./MainNav"

const styles = require("./Layout.scss")

type LayoutProps = {
  currentPath?: string
  pageTitle?: string
  children?: React.ReactNode
  footer?: React.ReactNode
  pageHeader?: React.ReactNode
  fullWidthContent?: boolean
  trimBottomOfCardToContent?: boolean
}

const Layout: React.SFC<LayoutProps> = ({
  pageTitle,
  currentPath,
  children,
  pageHeader,
  fullWidthContent = false,
  trimBottomOfCardToContent = false,
  footer,
}) => (
  <>
    <Head pageTitle={pageTitle} />
    <div className={styles.root}>
      <div className={styles.navigationBarContainer}>
        <MainNav currentPath={currentPath} />
      </div>
      <div
        className={classnames({
          [styles.page]: true,
          [styles.noBottomPadding]: fullWidthContent,
          [styles.noPageHeader]: !pageHeader,
        })}
      >
        <main>
          {pageHeader}
          {fullWidthContent ? (
            children
          ) : (
            <div
              className={classnames({
                [styles.contentContainer]: true,
                [styles.trimBottomOfCardToContent]: trimBottomOfCardToContent,
              })}
            >
              <div className={styles.content}>{children}</div>
            </div>
          )}
        </main>
      </div>
      <footer>
        <div className={styles.footerContainer}>{footer}</div>
      </footer>
    </div>
  </>
)

export default Layout
