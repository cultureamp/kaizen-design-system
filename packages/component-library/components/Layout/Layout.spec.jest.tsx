import React from "react"
import { render } from "@testing-library/react"
import Layout from "./Layout"

describe("Layout components", () => {
  it("renders the NavigationBar component correctly", () => {
    const { container } = render(<Layout.NavigationBar />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the Sidebar component correctly", () => {
    const { container } = render(<Layout.Sidebar />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the Header component correctly", () => {
    const { container } = render(<Layout.Header />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the Footer component correctly", () => {
    const { container } = render(<Layout.Footer />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the Toasts component correctly", () => {
    const { container } = render(<Layout.Toasts />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the Announcers component correctly", () => {
    const { container } = render(<Layout.Announcers />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the Layout component correctly", () => {
    const { container } = render(<Layout />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it("renders the Layout component's child components in the correct positions", () => {
    const { container } = render(
      <Layout>
        <Layout.NavigationBar>NavigationBar</Layout.NavigationBar>
        <Layout.Sidebar>Sidebar</Layout.Sidebar>
        <Layout.Header>Header</Layout.Header>
        <Layout.Footer>Footer</Layout.Footer>
        <Layout.Toasts>Toasts</Layout.Toasts>
        <Layout.Announcers>Announcers</Layout.Announcers>
        <div>My content</div>
      </Layout>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
