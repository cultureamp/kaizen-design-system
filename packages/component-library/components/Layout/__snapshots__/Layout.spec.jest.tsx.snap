// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout components renders the Announcers component correctly 1`] = `
<div
  aria-live="assertive"
  class="announcers"
/>
`;

exports[`Layout components renders the Footer component correctly 1`] = `
<footer
  class="footer"
/>
`;

exports[`Layout components renders the Header component correctly 1`] = `
<aside
  class="header"
/>
`;

exports[`Layout components renders the Layout component correctly 1`] = `
<div
  class="root"
>
  <div
    class="page"
  >
    <div
      class="body"
    >
      <div
        class="bodyInner"
      >
        <main
          class="content"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Layout components renders the Layout component's child components in the correct positions 1`] = `
<div
  class="root"
>
  <div
    class="navigationBar"
  >
    NavigationBar
  </div>
  <div
    aria-live="assertive"
    class="announcers"
  >
    Announcers
  </div>
  <div
    class="page"
  >
    <aside
      class="header"
    >
      Header
    </aside>
    <div
      aria-live="assertive"
      class="toasts"
    >
      Toasts
    </div>
    <div
      class="body"
    >
      <div
        class="bodyInner"
      >
        <div
          class="sidebar"
        >
          <div
            class="sidebarInner"
          >
            Sidebar
          </div>
        </div>
        <main
          class="content"
        >
          <div>
            My content
          </div>
        </main>
      </div>
    </div>
    <footer
      class="footer"
    >
      Footer
    </footer>
  </div>
</div>
`;

exports[`Layout components renders the NavigationBar component correctly 1`] = `
<div
  class="navigationBar"
/>
`;

exports[`Layout components renders the Sidebar component correctly 1`] = `
<div
  class="sidebar"
>
  <div
    class="sidebarInner"
  />
</div>
`;

exports[`Layout components renders the Toasts component correctly 1`] = `
<div
  aria-live="assertive"
  class="toasts"
/>
`;
